src/Equations/ConfinementTime.cpp

```diff
@@ -12,33 +17,65 @@ using namespace std;
 /**
  * Constructor
  */
+
 ConfinementTime::ConfinementTime(
 	FVM::UnknownQuantityHandler *u, EllipticalRadialGridGenerator *r,
-	IonHandler *ions, ConnectionLength* CL, len_t D_index
-) {
+	IonHandler *ions, ConnectionLength* CL, STREAM::EquationSystem *eqsys,  len_t D_index)
+{
     unknowns = u;
     radials  = r;
+    
     this->ions = ions;
     this->CL = CL;
+    this->eqsys = eqsys;
     this->D_index = D_index;
     
-    id_Tcold = unknowns->GetUnknownID(OptionConstants::UQTY_T_COLD);
-    id_Wi    = unknowns->GetUnknownID(OptionConstants::UQTY_WI_ENER);
-    id_Ni    = unknowns->GetUnknownID(OptionConstants::UQTY_NI_DENS);
+    
+    id_Tcold = unknowns->GetUnknownID(DREAM::OptionConstants::UQTY_T_COLD);
+    id_Wi    = unknowns->GetUnknownID(DREAM::OptionConstants::UQTY_WI_ENER);
+    id_Ni    = unknowns->GetUnknownID(DREAM::OptionConstants::UQTY_NI_DENS);
+    id_Ip    = unknowns->GetUnknownID(DREAM::OptionConstants::UQTY_I_P);
+    id_Ncold = unknowns->GetUnknownID(DREAM::OptionConstants::UQTY_N_COLD);
+    id_Efield= unknowns->GetUnknownID(DREAM::OptionConstants::UQTY_E_FIELD);
+    
+    type = (enum STREAM::OptionConstants::Conf_Time_type)eqsys->GetSettings()->GetInteger("eqsys/tau_perp/tau_perp"); //Setting the type of confinement time
+	mixedConfLaw = (bool)eqsys->GetSettings()->GetInteger("eqsys/tau_perp/mixte");
+    smoothless = (STREAM::OptionConstants::Conf_Time_smoothless)eqsys->GetSettings()->GetInteger("eqsys/tau_perp/smoothless");
+	smoothlessFunction = setSmoothless();
```

Comment:
would be nice to define smooth and smoothless....

Reply:
if I understand correctly the 'smoothless' thing is the way you mix the confinement laws. Currently we are going with the minimum based on literature. Since we are only using one, this option should disappear completely no?

Reply:
This has been removed